pkg/api/domains/domains_test.go

```diff
@@ -0,0 +1,273 @@
+			// WHEN a WLS domain is appended to the cluster and the Init() method is invoked
+			createAndInsertWLSDomain(tt)
+			// THEN the single domain with the expected name will be found in the Domains slice
+			assert.Equal(t, tt.listers, listerSet, "Wrong number of listers")
+			assert.Equal(t, 1, len(Domains), "Wrong number of Domains")
+			assert.Equal(t, "test3", Domains[0].Namespace, "Domain has the wrong namespace")
+		})
+	}
+}
+
+//TestReturnNoDomains ensures that an empty response is returned if no domains are available
+func TestReturnNoDomains(t *testing.T) {
+	// GIVEN an empty model binding pair
+	Domains = nil
+	emptyModelBindingPairs := make(map[string]*types.ModelBindingPair)
+	Init(controller.Listers{
+		ModelBindingPairs: &emptyModelBindingPairs,
+	})
+	// setup http request processing
+	response := httptest.NewRecorder()
+	request := httptest.NewRequest("GET", "/domains", nil)
+	// WHEN an HTTP request for the list of domains is submitted
+	ReturnAllDomains(response, request)
+	// THEN a HTTP OK response with an empty array body will be returned
+	assert.Equal(t, http.StatusOK, response.Code, "wrong response code")
+	var domains []Domain
+	err := json.Unmarshal(response.Body.Bytes(), &domains)
+	if err != nil {
+		t.Errorf("Error while reading response JSON: %s", err)
+	}
+	// assert empty response
+	assert.Empty(t, domains, "response should have no domains")
+}
+
+//TestReturnAllDomains ensures that an array of domains is returned
+func TestReturnAllDomains(t *testing.T) {
+	Domains = nil
+	clusters := testutil.GetTestClusters()
+	var clients kubernetes.Interface = fake.NewSimpleClientset()
+	modelBindingPairs := map[string]*types.ModelBindingPair{
+		"test-pair-1": getModelBindingPair(),
+	}
+	// setup the tests
+	tests := []struct {
+		name    string
+		listers controller.Listers
+	}{
+		{name: "testReturnAllDomains",
+			listers: testutilcontroller.NewControllerListers(&clients, clusters, &modelBindingPairs)},
+	}
+	// run the tests
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// GIVEN an array of clusters and a model binding pair that includes a reference to a WLS Domain
+			createAndInsertWLSDomain(tt)
+			// setup http request processing
+			response := httptest.NewRecorder()
+			request := httptest.NewRequest("GET", "/domains", nil)
+			// WHEN a request is submitted for a listing of domains
+			ReturnAllDomains(response, request)
+			// THEN the response with have an HTTP OK status (200) with a JSON array of domains in the response body
+			assert.Equal(t, http.StatusOK, response.Code, "wrong response code")
+			var domains []Domain
+			err := json.Unmarshal(response.Body.Bytes(), &domains)
+			if err != nil {
+				t.Errorf("Error while reading response JSON: %s", err)
+			}
+			// Validate that there is the correct number of domains, and the
+			// domains have the expected namespaces
+			assert.Equal(t, 1, len(domains), "Wrong number of domains in the list")
```

Comment:
this test is supposed to be testing multiple domains, can you add a second domain?

Reply:
done